### PR TITLE
feat(session replay): add logging to SR SDK

### PR DESCRIPTION
**Pull request by kwalker3690**
### Summary

This PR adds extra logs to the SR SDK lifecycle, as well as adds storage size data to the custom event we are sending along with our rrweb events.

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-TypeScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  no

### CODE REVIEW

File: packages/session-replay-browser/src/session-replay.ts [hunk 1]
```diff
@@ -406,19 +405,23 @@ export class SessionReplay implements AmplitudeSessionReplay {
     const { apiKey } = config;
     config.apiKey = `****${apiKey.substring(apiKey.length - 4)}`;
 
+    const storageSizeData = await getStorageSize();
+
     return {
       config,
       version: VERSION,
+      ...storageSizeData,
     };
```
Comment (lewgordon-amplitude) [inline packages/session-replay-browser/src/session-replay.ts:414]:
```suggestion
    return {
      ...storageSizeData,
      config,
      version: VERSION,
    };
```

This is probably unlikely, but I wouldn't want `storageSizeData` to potentially override config/version

File: packages/session-replay-browser/src/helpers.ts [hunk 2]
```diff
@@ -130,3 +137,17 @@ export const getServerUrl = (serverZone?: keyof typeof ServerZone): string => {
...
 };
+
+export const getStorageSize = async (): Promise<StorageData> => {
+  const globalScope = getGlobalScope();
+  if (globalScope) {
+    const storeEstimate: ChromeStorageEstimate = await globalScope.navigator.storage.estimate();
+    const totalStorageSize = storeEstimate.usage ? Math.round(storeEstimate.usage / KB_SIZE) : 0;
+    const percentOfQuota =
+      storeEstimate.usage && storeEstimate.quota
+        ? Math.round((storeEstimate.usage / storeEstimate.quota + Number.EPSILON) * 1000) / 1000
```
Comment (lewgordon-amplitude) [inline packages/session-replay-browser/src/helpers.ts:148]: Why the use of `Number.EPSILON` here?

File: packages/session-replay-browser/src/helpers.ts [hunk 3]
```diff
@@ -130,3 +137,17 @@ export const getServerUrl = (serverZone?: keyof typeof ServerZone): string => {
...
+        : 0;
+    return { totalStorageSize, percentOfQuota, usageDetails: JSON.stringify(storeEstimate.usageDetails) };
+  }
```
Comment (lewgordon-amplitude) [inline packages/session-replay-browser/src/helpers.ts:151]:
```suggestion
    const {usage, quota, usageDetails}: ChromeStorageEstimate = await globalScope.navigator.storage.estimate();
    const totalStorageSize = usage ? Math.round(usage / KB_SIZE) : 0;
    const percentOfQuota =
      usage && quota
        ? Math.round((usage / quota + Number.EPSILON) * 1000) / 1000
        : 0;
    return { totalStorageSize, percentOfQuota, usageDetails: JSON.stringify(usageDetails) };
```

Nit: I think this is slightly cleaner?

**Review by lewgordon-amplitude — APPROVED** [review]
Nice! I think this is going to be really useful to help debug 💪

File: packages/session-replay-browser/src/helpers.ts (hunk 2 above)
Comment (kwalker3690) [reply packages/session-replay-browser/src/helpers.ts:148]: It's a surprisingly controversial way of helping JS to round a number with floating points: https://stackoverflow.com/questions/11832914/how-to-round-to-at-most-2-decimal-places-if-necessary

**Pull request merged**